tests/unit/igraph_circulant.c

```diff
@@ -0,0 +1,195 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+
+#include "test_utilities.inc"
+
+int main() {
+
+    igraph_t graph, graph_test;
+    igraph_vector_int_t l;
+    igraph_bool_t iso;
+    igraph_integer_t i;
+
+    /* Testing invalid values */
+
+    /* n = -3, l = [1], undirected */
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
```

Comment:
Tip for when you want to create a small vector with specific elements:

Take a look at `igraph_vector_int_init_int_end()` and `igraph_vector_int_init_int()`. We can create the vector (1,2,3) as:

```
igraph_vector_int_init_int(&vec, 3 /* length */, 1,2,3 /* elements */);

/* or */

igraph_vector_int_init_int(&vec, -1 /* end marker */, 1,2,3, /* elements */ -1 /* the end */);
```

It's up to you what you use. This is just so you are aware of possibilities. There is no need to rewrite this test.

Reply:
Definitely alot cleaner, so I've gone ahead and updated the tests anyway. I've also changed the tests to use `shifts` instead of `l` to be consistent.